### PR TITLE
feat: prevent loss of an unsaved entry.

### PR DESCRIPTION
**Pull request by antoine-amara**
When creating a new entry, if a user click on an another entry to view
it, ask to him to confirm his action to prevent a data loss.

Update the onClick event on a list element (entries-list) to
be able to check the state of the app and display the dialog window
if it is needed. Create a redux selector to retrieve the mode field
to be able to check into the list if a new entry is opened.

Fixes: https://github.com/buttercup/buttercup-desktop/issues/634

### CODE REVIEW

File: locales/fa/base.json [hunk 1]
```diff
@@ -150,7 +150,8 @@
     "custom-fields-label-empty-info": "شما افزودن عنوان به یکی از فیلدهای شخصی را فراموش کردید.",
     "are-you-sure-question": "برای این کار اطمینان دارید؟",
     "displaying-entries": "در حال نمایش {{count}} سند",
-    "no-value": "بدون محتوی"
+    "no-value": "بدون محتوی",
+    "quit-unsave-entry": "سوف تفقد دخولك ، هل أنت متأكد؟"
```
Comment (sallar) [inline locales/fa/base.json]: Please remove the new translations except English. This is up to collaborators to add the correct translations. I see that you have used Google Translate for these which is not correct :)

File: src/renderer/components/archive/entries-list.js [hunk 2]
```diff
@@ -54,13 +56,25 @@ ListItemContent.propTypes = {
...
+}) => (
   <ListWrapper>
     <For each="entry" of={entries}>
       <ListItem
         key={entry.id}
         active={currentEntry && entry.id === currentEntry.id}
-        onClick={() => onSelectEntry(entry.id)}
+        onClick={() =>
+          !currentEntry && currentEntryMode === 'new'
+            ? showConfirmDialog(
```
Comment (sallar) [inline src/renderer/components/archive/entries-list.js]: This should be handled in the redux actions file. Search the project for `showConfirmDialog` usages to see how it's done in other places. It should not be handled inside the presentation components.

**Comment by antoine-amara** [comment]
Thanks for the review @sallar , I made the changes and I think I found the way to handle the check into the redux action. About translations, I keep english and french (because I'm french).

File: src/shared/actions/entries.js [hunk 3]
```diff
@@ -20,7 +20,27 @@ import {
...
 import { loadGroup } from '../../shared/actions/groups';
 
-export const selectEntry = createAction(ENTRIES_SELECTED);
+export const selectEntry = (
+  entryId,
+  currentEntry,
+  currentEntryMode
+) => async dispatch => {
+  try {
+    !currentEntry && currentEntryMode === 'new'
```
Comment (sallar) [inline src/shared/actions/entries.js]: You can get `currentEntry` and `currentEntryMode` here directly using `getState` and the selector functions, no need to pass them in from the UI component.

**Comment by antoine-amara** [comment]
Hello @sallar , yes I will push the changes soon, I'm on it.

**Comment by sallar** [comment]
@antoine-amara the tests are failing :) please have a look:
https://travis-ci.org/buttercup/buttercup-desktop/builds/458136631?utm_source=github_status&utm_medium=notification

**Comment by antoine-amara** [comment]
Hello, I see, sorry for that, I forget to delete an unused import. You prefer to have a new commit for the fix or I have to ammend my first commit ?

**Pull request merged**